apps/prairielearn/src/ee/pages/institutionAdminLti13/institutionAdminLti13.ts

```diff
@@ -0,0 +1,229 @@
+import { Router } from 'express';
+import asyncHandler = require('express-async-handler');
+import jose = require('node-jose');
+import { z } from 'zod';
+import { sortBy } from 'lodash';
+
+import error = require('@prairielearn/error');
+import { loadSqlEquiv, queryAsync, queryRows } from '@prairielearn/postgres';
+import { flash } from '@prairielearn/flash';
+
+import { getInstitution } from '../../lib/institution';
+import { InstitutionAdminLti13 } from './institutionAdminLti13.html';
+import { Lti13InstanceSchema } from '../../../lib/db-types';
+import { getCanonicalHost } from '../../../lib/url';
+import { config } from '../../../lib/config';
+
+const sql = loadSqlEquiv(__filename);
+const router = Router({ mergeParams: true });
+
+type LTI13Platform = {
+  platform: string;
+  display_order: number;
+  issuer_params?: object;
+  custom_fields?: object;
+}[];
+
+export type { LTI13Platform };
+
+// Middleware to check for feature and access
+router.use(
+  asyncHandler(async (req, res, next) => {
+    if (!res.locals.lti13_enabled) {
+      throw error.make(403, 'Access denied (feature not available)');
+    }
+    next();
+  }),
+);
+
+router.get(
+  '/:unsafe_lti13_instance_id?',
+  asyncHandler(async (req, res) => {
+    const institution = await getInstitution(req.params.institution_id);
+    const lti13Instances = await queryRows(
+      sql.select_instances,
+      {
+        institution_id: req.params.institution_id,
+      },
+      Lti13InstanceSchema,
+    );
+
+    const platform_defaults_hardcoded: LTI13Platform = [
+      {
+        platform: 'Unknown',
+        display_order: 0,
+        issuer_params: {},
+      },
+      {
+        platform: 'Canvas Production',
+        display_order: 10,
+        issuer_params: {
+          issuer: 'https://canvas.instructure.com',
+          jwks_uri: 'https://sso.canvaslms.com/api/lti/security/jwks',
+          token_endpoint: 'https://sso.canvaslms.com/login/oauth2/token',
+          authorization_endpoint: 'https://sso.canvaslms.com/api/lti/authorize_redirect',
+        },
+        custom_fields: {
+          uin: '$Canvas.user.sisIntegrationId',
+        },
+      },
+    ];
+
+    const platform_defaults = sortBy(
+      [...platform_defaults_hardcoded, ...config.lti13InstancePlatformDefaults],
+      ['display_order', 'platform'],
+    );
+
+    // Handle the / (no id passed case)
+    if (typeof req.params.unsafe_lti13_instance_id === 'undefined' && lti13Instances.length > 0) {
+      return res.redirect(`lti13/${lti13Instances[0].id}`);
+    }
+
+    const paramInstance = lti13Instances.find(
+      ({ id }) => id === req.params.unsafe_lti13_instance_id,
+    );
```

Comment:
Do we want to 404 if instance isn't found?

Reply:
Better route checking added in e58bb1c9